iosApp/Survey/Sources/Presentation/Modules/SurveySelection/AccountView/AccountView+DataSource.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AccountView+DataSource.swift
+//  Survey
+//
+//  Created by Bliss on 22/12/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Combine
+import KMPNativeCoroutinesCombine
+import Shared
+
+extension AccountView {
+
+    final class DataSource: ObservableObject {
+
+        private let viewModel: AccountViewModel
+        private let coordinator: AccountCoordinator
+
+        @Published private(set) var viewState = AccountViewState()
+        @Published var showingLoading = false
+
+        private var cancellables = Set<AnyCancellable>()
+
+        init(
+            coordinator: AccountCoordinator,
+            viewModel: AccountViewModel
+        ) {
+            self.viewModel = viewModel
+            self.coordinator = coordinator
+            createGuaranteedPublisher(
+                for: viewModel.viewStateNative,
+                fallback: AccountViewState()
+            )
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] value in
+                guard let self = self else { return }
```

Comment:
If you use the latest version of Xcode, then
```suggestion
                guard let self else { return }
```